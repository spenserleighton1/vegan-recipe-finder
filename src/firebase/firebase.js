import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore'

const config = {
    apiKey: "AIzaSyDfChryFf9ERaEQz2RgyyN4VzkPKAEzuqY",
    authDomain: "vegan-recipe-finder.firebaseapp.com",
    databaseURL: "https://vegan-recipe-finder.firebaseio.com",
    projectId: "vegan-recipe-finder",
    storageBucket: "",
    messagingSenderId: "1056656932624"
  };

firebase.initializeApp(config);

export const auth = firebase.auth();
export const firestore = firebase.firestore();
export const docRef = firestore.doc("users/savedRecipes");

export default { auth, docRef };