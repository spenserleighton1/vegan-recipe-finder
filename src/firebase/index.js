import * as auth from './auth';
import * as firebase from './firebase';

import { docRef } from './firebase'


export {
  auth,
  firebase,
  docRef
};