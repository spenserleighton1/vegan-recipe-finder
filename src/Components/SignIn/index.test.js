import React from 'react';
import { shallow } from 'enzyme';
import { SignInForm } from './';
import { auth } from '../../firebase'

jest.mock('../../firebase');

describe('SignInForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignInForm />)
  })
  it('should have default state of email, password and error', () => {
    const actual = wrapper.state()
    const expected = {"email": "", "error": null, "password": ""}

    expect(actual).toEqual(expected)
  })

  it.skip('should add a user to firebase when onSubmit is called', () => {
    const mockEvent = { preventDefault: jest.fn() }
    wrapper.instance().onSubmit(mockEvent)

    console.log(auth)
  })

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: {name: 'email', value: 'sleepy' }}

    wrapper.instance().handleChange(mockEvent)

    const actual = wrapper.state('email')

    expect(actual).toEqual('sleepy')
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})