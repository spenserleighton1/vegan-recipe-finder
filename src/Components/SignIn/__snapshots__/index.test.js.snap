// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignInPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          SignIn
</h2>,
        <form
          onSubmit={[Function]}
>
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
                    value=""
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign In
          </button>
          <p>
                    Don't have an account?
                    <Link
                              replace={false}
                              to="/signUp"
                    >
                               Sign Up
                    </Link>
          </p>
          <p />
</form>,
      ],
      "className": "sign-in",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SignIn",
        },
        "ref": null,
        "rendered": "SignIn",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign In
</button>,
            <p>
              Don't have an account?
              <Link
                            replace={false}
                            to="/signUp"
              >
                             Sign Up
              </Link>
</p>,
            <p />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Don't have an account?",
                <Link
                  replace={false}
                  to="/signUp"
>
                   Sign Up
</Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Don't have an account?",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Sign Up",
                  "replace": false,
                  "to": "/signUp",
                },
                "ref": null,
                "rendered": " Sign Up",
                "type": [Function],
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            SignIn
</h2>,
          <form
            onSubmit={[Function]}
>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Email Address"
                        type="text"
                        value=""
            />
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Sign In
            </button>
            <p>
                        Don't have an account?
                        <Link
                                    replace={false}
                                    to="/signUp"
                        >
                                     Sign Up
                        </Link>
            </p>
            <p />
</form>,
        ],
        "className": "sign-in",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SignIn",
          },
          "ref": null,
          "rendered": "SignIn",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
/>,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                disabled={true}
                type="submit"
>
                Sign In
</button>,
              <p>
                Don't have an account?
                <Link
                                replace={false}
                                to="/signUp"
                >
                                 Sign Up
                </Link>
</p>,
              <p />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Don't have an account?",
                  <Link
                    replace={false}
                    to="/signUp"
>
                     Sign Up
</Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Don't have an account?",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " Sign Up",
                    "replace": false,
                    "to": "/signUp",
                  },
                  "ref": null,
                  "rendered": " Sign Up",
                  "type": [Function],
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
